core/commands/block.go

```diff
@@ -121,14 +122,17 @@ var blockPutCmd = &cmds.Command{
 		ShortDescription: `
 'ipfs block put' is a plumbing command for storing raw IPFS blocks.
 It reads from stdin, and <key> is a base58 encoded multihash.
+
+By default CIDv0 is going to be generated. Setting 'mhtype' to anything other
+than 'sha2-256' or format to anything other than 'v0' will result in CIDv1.
 `,
 	},
 
 	Arguments: []cmdkit.Argument{
 		cmdkit.FileArg("data", true, false, "The data to be stored as an IPFS block.").EnableStdin(),
 	},
 	Options: []cmdkit.Option{
-		cmdkit.StringOption("format", "f", "cid format for blocks to be created with.").WithDefault("v0"),
+		cmdkit.StringOption("format", "f", "cid format for blocks to be created with.").WithDefault(""),
```

Comment:
I wouldn't specify a default at all. As it is, passing `-f ""` will be considered valid even though it makes no sense.

If you don't specify a default, you can tell if the parameter was passed with `format, formatSet := req.Options["format"].(string)`.